cmd/medius/images/push.go

```diff
@@ -200,9 +200,11 @@ func (b *buildAndPublish) getArtifact(artifactInfo *api.ArtifactDetails) (string
 		return "", b.Ctx.Err()
 	}
 
-	checksum := artifactReader.Checksum()
-	if checksum != artifactInfo.SHA256Sum {
-		return "", fmt.Errorf("expected checksum %q but got %q", artifactInfo.SHA256Sum, checksum)
+	checksum256, checksum512 := artifactReader.Checksum()
```

Comment:
Can you put the checksum type into `api.ArtifactDetails` instead and pass it to `b.Getter.GetWithChecksumAndContext`. The checksum returned by `artifactReader.Checksum()` could then simply be compared to `artifactInfo.Checksum`.

Reply:
I'm not sure If I'm following. The idea is to compare the checksum present in "api.ArtifactDetails", which is retrieved from ".json" in the repo, compute our own and check if they match.

Reply:
I'm asking you not to return two possible checksums from `artifactReader.Checksum()`. If you put the expected type of the checksum into `api.ArtifactDetails`, then `artifactReader.Checksum()` can return only a single checksum in the expected format.

Reply:
Ah I see. Thanks for the clarification. Done.